code/modules/mob/mob.dm

```diff
@@ -331,6 +331,8 @@
 		return FALSE
 
 	//Each inventory slot can have more than one define associated with it. This refines the slot down to the item types actually associated with the define.
+	if(slot == SLOT_IN_UNIFORM_HOLSTER && (!(istype(I, /obj/item/clothing/under) || istype(I, /obj/item/weapon) || istype(I, /obj/item/storage/holster))))
```

Comment:
Human's get_item_by_slot returns the uniform if you ask for the uniform holster, but you're checking whether it's a weapon or holster here.